contracts/gauge/Gauge.sol

```diff
@@ -0,0 +1,217 @@
+    function changeDistribution(address _distribution) external onlyGovernance {
+        DISTRIBUTION = _distribution;
+    }  
+
+    // total supply of our lp tokens in the gauge (e.g. AC4D tokens present)
+    function totalSupply() external view returns (uint256) {
+        return _totalSupply;
+    }
+
+    // balance of lp tokens that user has in the gauge (e.g. amount of AC4D a user has)
+    function balanceOf(address account) external view returns (uint256) {
+        return _balances[account];
+    }
+
+    function lastTimeRewardApplicable() public view returns (uint256) {
+        return Math.min(block.timestamp, periodFinish);
+    }
+
+    // how many of our reward tokens is the user receiving per lp token
+    // (e.g. how many teddy or axial is received per AC4D token)
+    function rewardPerToken(uint256 tokenIndex) public view returns (uint256) {
+        if (_totalSupply == 0) {
+            return rewardPerTokenStored[rewardTokens[tokenIndex]];
+        }
+        return
+            rewardPerTokenStored[rewardTokens[tokenIndex]].add(
+                lastTimeRewardApplicable().sub(lastUpdateTime).mul(rewardRates[rewardTokens[tokenIndex]]).mul(1e18).div(derivedSupply)
+            );
+    }
+
+    function derivedBalance(address account) public view returns (uint) {
+        uint _balance = _balances[account];
+        uint _derived = _balance.mul(40).div(100);
+        uint _adjusted = (_totalSupply.mul(XAXIAL.balanceOf(account)).div(XAXIAL.totalSupply())).mul(60).div(100);
+        return Math.min(_derived.add(_adjusted), _balance);
+    }
+
+    function kick(address account) public {
+        uint _derivedBalance = derivedBalances[account];
+        derivedSupply = derivedSupply.sub(_derivedBalance);
+        _derivedBalance = derivedBalance(account);
+        derivedBalances[account] = _derivedBalance;
+        derivedSupply = derivedSupply.add(_derivedBalance);
+    }
+
+    // ??? REWARDS???
+    // it is saying the amoun t of rewardtokens already paid for xisting reward tokens 
+    // rewards? the amount of reward tokens/ the amount of rewards tokens for the reward token
+    function earned(address account, uint256 tokenIndex) public view returns (uint256) {
+        return derivedBalances[account].mul(rewardPerToken(tokenIndex).sub(userRewardPerTokenPaid[account][rewardTokens[tokenIndex]])).div(1e18).add(rewards[account][rewardTokens[tokenIndex]]);
+    }
+
+    function getRewardForDuration(uint256 tokenIndex) external view returns (uint256) {
+        return rewardRates[rewardTokens[tokenIndex]].mul(DURATION);
+    }
+
+    function _deposit(uint amount, address account) internal nonReentrant updateReward(account) {
+        require(amount > 0, "Cannot stake 0");
+        _totalSupply = _totalSupply.add(amount);
+        _balances[account] = _balances[account].add(amount);
+        emit Staked(account, amount);
+        TOKEN.safeTransferFrom(account, address(this), amount);
```

Comment:
See for the interface of local `SafeERC20` https://github.com/Snowball-Finance/axial-protocol/blob/main/contracts/libraries/SafeERC20.sol#L31

```

contracts/gauge/Gauge.sol:145:9: TypeError: Wrong argument count for function call: 3 arguments given but expected 2.
        TOKEN.safeTransferFrom(account, address(this), amount);
        ^----------------------------------------------------^

contracts/gauge/Gauge.sol:194:9: TypeError: Wrong argument count for function call: 3 arguments given but expected 2.
        AXIAL.safeTransferFrom(DISTRIBUTION, address(this), reward);
        ^---------------------------------------------------------^
 
 ```